securedrop/source_app/__init__.py

```diff
@@ -79,7 +84,7 @@ def handle_csrf_error(e):
     app.jinja_env.trim_blocks = True
     app.jinja_env.lstrip_blocks = True
     app.jinja_env.globals['version'] = version.__version__
-    if getattr(config, 'CUSTOM_HEADER_IMAGE', None):
+    if config.CUSTOM_HEADER_IMAGE:
```

Comment:
to keep things simple, SDConfig should not automagically change the semantic of attributes (in this case it means CUSTOM_HEADER_IMAGE exists even if not defined in config.py).

Reply:
This is related to [Truth value testing](https://docs.python.org/3/library/stdtypes.html#truth-value-testing). 

We know that there is an attribute of our config named `CUSTOM_HEADER_IMAGE`, but  use it only if it has a `Truth` value, means someone set it up  using config file.

In this PR I also clearing up the related code which are doing getattr or setattr (except the test cases).  Pushing another update in some time.

Reply:
such a change must be in a separate commit, with the explained rationale and benefits so it can be properly reviewed. Changing the semantics in the same commit a refactor is done is harder to review and may lead to subtle errors.

Reply:
Correct, but this has to be inside of that commit. As if we kept the old `hasattr` call, it would find the `CUSTOM_HEADER_IMAGE` all the time, and the would execute the next two lines. 

Now, it will only do if there is a proper value. The rest of any such calls are replaced in a newer commit in the PR.